### PR TITLE
Feat: apollo small enhancements

### PR DESCRIPTION
**Pull request by Gohnnyman**
<h1> What's new </h1>

* `requestId` removed from `PriceFeedRequested` event in `ApolloCoordinator`
* `dataFeedId` in `PriceFeedRequested` event in `ApolloCoordinator` is not indexed now 
* some visibility modifiers changed to `external` duo to it's lower gas usage. [info is here](https://ethereum.stackexchange.com/questions/19380/external-vs-public-best-practices)

---

## PR-Codex overview
The focus of this PR is to update ApolloCoordinator and related contracts, replacing ApolloCoordinatorV2 with ApolloCoordinator.

### Detailed summary
- Replaced `ApolloCoordinatorV2` with `ApolloCoordinator` in multiple contracts.
- Added `requestRandomFeed` function to `ApolloCoordinator`.
- Renamed `PriceFeedRequested` event to `DataFeedRequested` and added `RandomFeedRequested` event.
- Updated `ApolloConsumerExample` to use `ApolloReceiver`.
- Refactored `OrallyMulticall` contract for better readability.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/interfaces/IApolloCoordinatorV2.sol [hunk 1]
```diff
@@ -12,6 +11,6 @@ interface IApolloCoordinatorV2 {
     function requestDataFeed(string memory dataFeedId, uint256 callbackGasLimit) external;
 
     event PriceFeedRequested(
-        string indexed dataFeedId, uint256 callbackGasLimit, address indexed requester
+        string dataFeedId, uint256 callbackGasLimit, address indexed requester
```
Comment (verkhohliad) [inline src/interfaces/IApolloCoordinatorV2.sol]: why removed `indexed` here?

File: src/interfaces/IApolloCoordinatorV2.sol [hunk 2]
```diff
@@ -3,7 +3,6 @@ pragma solidity 0.8.20;
 
 interface IApolloCoordinatorV2 {
     struct PriceFeedRequest {
```
Comment (verkhohliad) [inline src/interfaces/IApolloCoordinatorV2.sol]: actually we already don't need this struct at in V2, forget to remove

File: src/interfaces/IApolloCoordinator.sol [hunk 3]
```diff
@@ -3,13 +3,14 @@ pragma solidity 0.8.20;
...
     struct PriceFeedRequest {
-        uint256 requestId; // Unique identifier for the request
         string dataFeedId; // Identifier for the type of data requested
         uint256 callbackGasLimit; // Gas limit for the callback transaction
         address requester; // Address of the requesting contract
     }
 
     event PriceFeedRequested(
-        uint256 indexed requestId, string indexed dataFeedId, uint256 callbackGasLimit, address indexed requester
+        string dataFeedId,
```
Comment (verkhohliad) [inline src/interfaces/IApolloCoordinator.sol]: we don't need to maintain already legacy version of Apollo

File: src/apollo/ApolloCoordinator.sol [hunk 4]
```diff
@@ -20,26 +20,28 @@ contract ApolloCoordinator is IApolloCoordinator {
      * @param dataFeedId The identifier of the data feed being requested.
      * @param callbackGasLimit The gas limit for the callback transaction.
      */
-    function requestDataFeed(string memory dataFeedId, uint256 callbackGasLimit) public {
-        uint256 id = _getRequestId();
-        requests[id] = PriceFeedRequest(id, dataFeedId, callbackGasLimit, msg.sender);
-        emit PriceFeedRequested(id, dataFeedId, callbackGasLimit, msg.sender);
+    function requestDataFeed(
```
Comment (verkhohliad) [inline src/apollo/ApolloCoordinator.sol:19]: same as for interface, we don't need to support legacy version, it'll be removed

File: src/OrallyMulticall.sol [hunk 5]
```diff
@@ -46,7 +52,7 @@ contract OrallyMulticall is OrallyConsumer {
         return returnData;
     }
 
-    function multitransfer(Transfer[] calldata transfers) public payable onlyExecutor {
+    function multitransfer(Transfer[] calldata transfers) external payable onlyExecutor {
```
Comment (verkhohliad) [inline src/OrallyMulticall.sol:55]: can you please specify why this changed? cause it'll require do redeploy some infra contracts on already deployed chains

File: script/Deploy_1.s.sol [hunk 6]
```diff
@@ -10,12 +10,14 @@ import {OrallyMulticall} from "src/OrallyMulticall.sol";
...
 
+import {ApolloCoordinatorV2} from "src/apollo/ApolloCoordinatorV2.sol";
+
 contract Deploy_1 is Script {
     address constant pmaAddress = 0x05C3F2A3Ae0b7f3775044EEFED8a864c47125F19;
     address constant sybilAddress = 0x60825063CB0f4EF508854Ad4913f3a6de86B3807;
 
     // different between chains, could be added in different deployment script () on already deployed infrastructure
-    address constant amaAddress;
+    address constant amaAddress = 0xA40a0617BA2b446b0300eeD8B7D8A4C790216314;
```
Comment (verkhohliad) [inline script/Deploy_1.s.sol]: I removed concrete address here for reason, do not to forget to change on correct address of chain during deploy

File: script/Deploy_1.s.sol [hunk 7]
```diff
@@ -10,12 +10,14 @@ import {OrallyMulticall} from "src/OrallyMulticall.sol";
 import {ApolloCoordinatorV2} from "src/apollo/ApolloCoordinatorV2.sol";
 import {OrallyPriceFeed} from "src/examples/OrallyPriceFeed.sol";
 
+import {ApolloCoordinatorV2} from "src/apollo/ApolloCoordinatorV2.sol";
```
Comment (verkhohliad) [inline script/Deploy_1.s.sol]: double import, check line 10

Comment (Gohnnyman) [reply script/Deploy_1.s.sol]: Removed

File: script/Deploy_1.s.sol (hunk 6 above)
Comment (Gohnnyman) [reply script/Deploy_1.s.sol]: forge doesn't compile anything if there any errors in any contract  :(

File: src/OrallyMulticall.sol (hunk 5 above)
Comment (Gohnnyman) [reply src/OrallyMulticall.sol:55]: Due to [this](https://ethereum.stackexchange.com/questions/19380/external-vs-public-best-practices) answer on stackexchange. In two words, in external functions solidity doesn't need to copy arguments from calldata to memory, as this func won't be called from other func inside this contract.

File: src/interfaces/IApolloCoordinatorV2.sol (hunk 2 above)
Comment (Gohnnyman) [reply src/interfaces/IApolloCoordinatorV2.sol]: Removed

File: src/interfaces/IApolloCoordinatorV2.sol (hunk 1 above)
Comment (Gohnnyman) [reply src/interfaces/IApolloCoordinatorV2.sol]: Indexed string results in abi-encoded value in output. Meaning if this string was indexed, in the event logs we would see not `"BTC/USD"`, but keccak256("BTC/USD") = `"0xee62665949c883f9e0f6f002eac32e00bd59dfe6c34e92a91c37d6a8322d6489"`

**Review by verkhohliad — APPROVED** [review]

**Review by verkhohliad — APPROVED** [review]

**Pull request merged**